### ISSUE TITLE
SetUpstream fail with error "bad argument #2 to 'encode' (table expected, got boolean)"

### ISSUE THREAD

**Issue by daverck** [Kong/go-pdk#114]
kong 2.8.1.1
go-pdk v0.8.0

I made a go plugin which use the SetUpstream method to select an existing upstream.

```go
errSetUpstream := kong.Service.SetUpstream(upstream)
```

The SetUpstream method is called during the access phase.
The plugin is activated for a specific route.
The upstream has only one target (xxx.xxx.xxx.xxx:8080)
The api on that target run fine.

I get the following error:

> 2022/06/23 06:46:09 [debug] 381368#0: *841963 [kong] pb_rpc.lua:260 before set upstream
> 2022/06/23 06:46:09 [error] 381368#0: *841963 [kong] init.lua:317 [kong-ls-plugin] ...cal/share/lua/5.1/kong/runloop/plugin_se
> rvers/pb_rpc.lua:266: bad argument #2 to 'encode' (table expected, got boolean), client: 172.16.35.41, server: kong, request: 
> "GET /api/v1/me HTTP/1.1", host: "xxxxxxxxxxxxxxxxxxxxxxxx", referrer: "xxxxxxxxxxxxxxxxxxxxxxx/home"
> 2022/06/23 06:46:09 [info] 381367#0: *48 [kong-ls-plugin:381397] 2022/06/23 06:46:09 write unix /usr/local/kong/kong-ls-plugin
> .socket->@: write: broken pipe, context: ngx.timer
> 2022/06/23 06:46:14 [debug] 381367#0: *842031 [lua] init.lua:295: [cluster_events] polling events from: 1655965268.725
> 2022/06/23 06:46:14 [info] 381368#0: *841963 client 172.16.35.41 closed keepalive connection
> 2022/06/23 06:46:14 [debug] 381367#0: *842047 [lua] init.lua:223: flush_counters(): [counters] pid 381367 acquired lock
> 2022/06/23 06:46:14 [debug] 3/home/dserck/kong-ls-plugin/proof/proof.go81370#0: *842048 [lua] init.lua:256: flush_counters(): [counters] flush done

Is this a bug ?
The same plugin in python works fine.

<details><summary>I stripped down my plugin to only this SetUpstream function call, and it still gives me this error.</summary>
<p>

```go
package main

import (
	"fmt"

	"github.com/Kong/go-pdk"
	"github.com/Kong/go-pdk/server"
)

func main() {
	server.StartServer(New, Version, Priority)
}

var Version = "0.1"
var Priority = 1

type Config struct {
	Message string
}

func New() interface{} {
	return &Config{}
}

func (conf Config) Access(kong *pdk.PDK) {
	kong.Log.Debug("----- proof -----")
	var upstream = "setupstream-test"
	kong.Log.Debug(fmt.Sprintf("upstream: %s", upstream))
	kong.Log.Debug("before set upstream")
	errSetUpstream := kong.Service.SetUpstream(upstream)
	kong.Log.Debug("after set upstream")
	if errSetUpstream != nil {
		kong.Log.Debug(fmt.Sprintf("kong set upstream err %s", errSetUpstream.Error()))
	}

}
```

</p>
</details>

**Comment by StarlightIbuki** [comment]
The issue seems to be caused by the incorrect typing of the return value. The return value is marked to return Empty, while it can return a flag and a possible error.

**Comment by gszr** [comment]
@daverck,

Thank you for the report; I have opened the following PRs fixing the issue:
- https://github.com/Kong/kong/pull/12727
- https://github.com/Kong/go-pdk/pull/191

I will keep this issue open until those are merged and I release a new version of the Go PDK.

**Comment by gszr** [comment]
The fix has been merged - and is part of the `v0.11.0` release of the Go PDK.
If you'd like to test and leverage the fix, pick the proto file [from Kong master](https://github.com/Kong/kong/blame/master/kong/include/kong/pluginsocket.proto) and give it a try.
